src/Microsoft.PowerShell.Commands.Utility/commands/utility/CoreCLR/TestConnectionCommand.cs

```diff
@@ -0,0 +1,224 @@
+            }
+            set
+            {
+                this.count = value;
+            }
+        }
+
+        /// <summary>Time in seconds between pings</summary>
+        [Parameter(ParameterSetName = "Default",
+            Mandatory = false,
+            HelpMessage = "Time in seconds between pings")]
+        [ValidateRange(1, 60)]
+        public int Delay
+        {
+            get
+            {
+                return this.delay;
+            }
+            set
+            {
+                this.delay = value;
+            }
+        }
+
+        /// <summary>Return only boolean value representing success/failure</summary>
+        [Parameter(ParameterSetName = "Default",
+            Mandatory =false,
+            HelpMessage ="Return only boolean value representing success/failure")]
+        public SwitchParameter Quiet
+        {
+            get
+            {
+                return (SwitchParameter)this.quiet;
+            }
+            set
+            {
+                this.quiet = (bool)value;
+            }
+        }
+#endregion
+#region Meat of the class
+        /// <summary> Create the object </summary>
+        protected override void ProcessRecord()
+        {
+            bool quietReply = false;
+            try
+            {
+                IPHostEntry ipHostEntry = Dns.GetHostEntryAsync(this.destination).Result;
+            
+            //If we got this far, somethings going good
+                try
+			    {
+				    for (int i = 0; i < this.count; i++)
+				    {
+					    Ping sender = new Ping();
+					    PingReply reply = sender.SendPingAsync(this.destination, (timeout * 1000), new byte[] { 0xff }, new PingOptions(this.timetolive, true)).Result;
+                    
+					    //If things are good then set our bool to true, otherwise leave it false or whatever it currently is
+					    if (reply.Status == IPStatus.Success)
+					    {
+						    quietReply = true;
+					    }
+
+					    //We want to be chatty
+					    if (!this.quiet)
+					    {
+						    WriteObject(new PingResponse(reply.Status.ToString(), reply.Address.ToString(), reply.RoundtripTime));
+					    }
+				    }
+				    //If we are passing -quiet we only want the one final pass/fail, so here it is
+				    if (this.quiet)
+				    {
+					    WriteObject(quietReply);
+				    }
+			    }
+			    catch (PingException pex)
+			    {
+				    WriteError(new ErrorRecord(pex, pex.Data.ToString(), ErrorCategory.NotSpecified, pex.Source));
+			    }
+			    catch (NullReferenceException nrefex)
+			    {
+				    WriteError(new ErrorRecord(nrefex, nrefex.Message, ErrorCategory.ObjectNotFound, nrefex.Source));
+			    }
+			    catch
+			    {
+				    //WriteError(new ErrorRecord(ex, ex.HResult.ToString(), ErrorCategory.NotSpecified, ex.Source));
+                    String errObject = this.destination + " not able to be resolved.\nPlease check the address and try again";
+                    WriteObject(errObject);

```

Comment:
>WriteObject [](start = 20, length = 11)

should this use WriteError, rather than just writing this string? It seems like this is an error condition and should be using WriteError